### PR TITLE
tests: Bluetooth: Run Audio BSIM test on nrf5340bsim

### PR DESCRIPTION
**Pull request by Thalley**
Add support for running the audio BSIM tests on the nrf5340bsim board.

This expands the tests we run in CI.

To support all tests with the hci_ipc sample for the netcore, it is updated to support a bit more while still fitting on the netcore on the nRF5340 series.

The behavior of advertisement seems to differ a little bit, which required the bap_unicast_server_test.c to update how it restarts advertisements.

Two broadcast tests that are passing on the nRF52 BSIM is also failing due an assert. This should be investigated at a later time.

### CODE REVIEW

File: tests/bsim/bluetooth/audio/test_scripts/cap_broadcast_ac_14.sh [hunk 1]
```diff
@@ -41,8 +41,12 @@ Execute_AC_14 24_1_1
 Execute_AC_14 24_2_1
 Execute_AC_14 32_1_1
 Execute_AC_14 32_2_1
-Execute_AC_14 441_1_1
-Execute_AC_14 441_2_1
+# ASSERTION FAIL [err == ((isoal_status_t) 0x00) || err == ((isoal_status_t) 0x04)]
```
Comment (aescolar) [inline tests/bsim/bluetooth/audio/test_scripts/cap_broadcast_ac_14.sh:44]: Just as a note: if this is failing only for the 5340, you could if it out in that case, and still run it for the 52.

**Review by aescolar — DISMISSED** [review]
Just a note, no complaints from me.

**Comment by jori-nordic** [comment]
> This should be investigated at a later time

Could you create an issue for tracking?

File: tests/bsim/bluetooth/audio/test_scripts/cap_broadcast_ac_14.sh (hunk 1 above)
Comment (Thalley) [reply tests/bsim/bluetooth/audio/test_scripts/cap_broadcast_ac_14.sh:44]: It is only failing for the 53 series. However, it's not a big deal to disable this particular test for the 52 series.

**Comment by Thalley** [comment]
> > This should be investigated at a later time
> 
> Could you create an issue for tracking?

Yup: https://github.com/zephyrproject-rtos/zephyr/issues/77810

**Comment by Thalley** [comment]
Slightly updated the `tests/bsim/bluetooth/audio/overlay-bt_ll_sw_split.conf` file so that it's more in sync with `samples/bluetooth/hci_ipc/nrf5340_cpunet_iso-bt_ll_sw_split.conf`

**Review by kruithofa — APPROVED** [review]

**Comment by Thalley** [comment]
@aescolar please re-review :)

**Review by aescolar — APPROVED** [review]

**Pull request merged**